qemu/tests/qemu_guest_agent.py

```diff
@@ -231,6 +233,7 @@ def setup(self, test, params, env):
             vm = self.env.get_vm(params["main_vm"])
             vm.verify_alive()
             self.vm = vm
+        if start_vm != "no":
```

Comment:
Is that means `start_vm == "yes"`, which is a same judgement with line 232, should be unnecessary?

Reply:
Line 232 means "not self.vm and start_vm == yes", Line 236 means only start_vm == yes. They are not the same. When start_vm == no, below line 236 should not be execucated.

Reply:
I guess what you want is:
```
if start_vm == "yes":
    if not self.vm:
        self.vm = self.env.get_vm(params["main_vm"])
        self.vm.verify_alive()
    session = self._get_session(params, self.vm)
    ......
```
Then you don't need to judge it twice

Reply:
Why not initiate self.vm in init function? You won't need judge self.vm exist or not everywhere therefore.

Reply:
yes, the above judgement is correct. Line 232-235 is written by others, so I just added my code in previous code. Maybe I can put it in init.

Reply:
I just checked gagent_check_fstrim method, we cann't put self.vm in init. The changes will influence the method. Because the method will get self.vm in setup().

Reply:
Then just modify it...

Reply:
If put self.vm in init, 
start_vm = self.params["start_vm"]                                       
if start_vm == "yes":                                                    
            vm = self.env.get_vm(params["main_vm"]) 
            self.vm = vm                             
It still needs to judgement, just a different place. There is not obvious improvement after modifying parent class with the risk of influenting other cases.  So I will modify it like you said at first.
if start_vm == "yes":
    if not self.vm:
        self.vm = self.env.get_vm(params["main_vm"])
        self.vm.verify_alive()
    session = self._get_session(params, self.vm)